[NAB-375] Variables with no init values referenced in arcs will cause arcs not to load

camperko opened the pull request:
# Description

Arcs which referenced data has non positive integer value will be displayed/represented with value 0 in edit + simulation mode

Fixes [NAB-375] 

## Dependencies

No new dependencies were introduced

### Third party dependencies

No new dependencies were introduced

### Blocking Pull requests

Blocking by [PF-79]

## How Has Been This Tested?

Manually tested in local dev env in running application

| Name                | Tested on |
|---------------------| --------- |
| OS                  |     Linux Mint 21      |
| Runtime             |      Node 20.13.1      |
| Dependency Manager  |    npm  10.8.0    |
| Framework version   |    Angular 13.3.1       |
| Run parameters      |           |
| Other configuration |           |


# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @...
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [ ] I have checked my contribution with code analysis tools:
    - [ ] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-builder)
    - [ ] [Snyk](https://app.snyk.io/org/netgrif)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides


[NAB-375]: https://netgrif.atlassian.net/browse/NAB-375?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PF-79]: https://netgrif.atlassian.net/browse/PF-79?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

camperko committed:
[NAB-375] Variables with no init values referenced in arcs will cause arcs not to load

- arcs which referenced data has non integer value will be displayed/represented with value 0 in edit + simulation mode

mazarijuraj commented on src/app/modeler/services/model/model.service.ts:
```diff
@@ -483,8 +483,10 @@ export class ModelService {
             if (referencedData.init.value) {
-                return Number(referencedData.init.value);
-                // TODO: NAB-326 check if isFinite and >= 0
+                if (/^[1-9]\d*$/.test(referencedData.init.value.trim())) {
```
Use static function from ImportUtils

mazarijuraj commented on src/app/modeler/simulation-mode/simulation-mode.service.ts:
```diff
@@ -87,7 +87,13 @@ export class SimulationModeService extends CanvasModeService<SimulationTool> {
-                .map(a => [a.reference, Number.parseInt(model.getData(a.reference).init?.value, 10) || 0]));
+                .map(a => {
+                    const data = model.getData(a.reference);
+                    if (!!data.init && !!data.init.value && /^[1-9]\d*$/.test(data.init.value)) {
```
Use static function from ImportUtils

tuplle committed:
[NAB-375] Variables with no init values referenced in arcs will cause arcs not to load

- Improved usage of ImportUtils.isInitValueNumber to replace regex checks for initialization values.
